pkg/daemon/ceph/client/osd.go

```diff
@@ -63,6 +64,76 @@ type OSDDump struct {
 		Up  json.Number `json:"up"`
 		In  json.Number `json:"in"`
 	} `json:"osds"`
+	Flags          string              `json:"flags"`
+	CrushNodeFlags map[string][]string `json:"crush_node_flags"`
+}
+
+// IsFlagSet checks if an OSD flag is set
+func (dump *OSDDump) IsFlagSet(checkFlag string) bool {
+	flags := strings.Split(dump.Flags, ",")
+	for _, flag := range flags {
+		if flag == checkFlag {
+			return true
+		}
+	}
+	return false
+}
+
+// IsFlagSetOnCrushUnit checks if an OSD flag is set on specified Crush unit
+func (dump *OSDDump) IsFlagSetOnCrushUnit(checkFlag, crushUnit string) bool {
+	for unit, list := range dump.CrushNodeFlags {
+		if crushUnit == unit {
+			for _, flag := range list {
+				if flag == checkFlag {
+					return true
+				}
+			}
+		}
+	}
+	return false
+}
+
+// UpdateFlagOnCrushUnit checks if the flag is in the desired state and sets/unsets if it isn't. Mitigates redundant calls
+// it returns true if the value was changed
+func (dump *OSDDump) UpdateFlagOnCrushUnit(context *clusterd.Context, set bool, clusterName, crushUnit, flag string) (bool, error) {
+	flagSet := dump.IsFlagSetOnCrushUnit(flag, crushUnit)
+	if flagSet && !set {
+		err := UnsetFlagOnCrushUnit(context, clusterName, crushUnit, flag)
+		if err != nil {
+			return true, err
+		}
+		return true, nil
+	}
+	if !flagSet && set {
+		err := SetFlagOnCrushUnit(context, clusterName, crushUnit, flag)
+		if err != nil {
+			return true, err
+		}
+		return true, nil
+	}
+	return false, nil
+}
+
+// SetFlagOnCrushUnit sets the specified flag on the crush unit
+func SetFlagOnCrushUnit(context *clusterd.Context, clusterName, crushUnit, flag string) error {
+	args := []string{"osd", "set-group", flag, crushUnit}
```

Comment:
This new flag appeared in 14.2.2, we must make sure we run at least this version

Reply:
@rohantmp Since we need this version, when deciding whether to enable PDBs with the `managePodBudgets` setting, we would avoid starting the controllers if the ceph version isn't sufficient. For example code search for `cephVersion.IsAtLeast`.

Reply:
Can we keep the PDB stuff, and just disable the `noout` stuff if the version is not sufficient for the flag? My unpushed fix is to log errors, not update noout, and proceed as usual.